usr.bin/truss/syscalls.c

```diff
@@ -1024,8 +1025,26 @@ get_syscall(struct threadinfo *t, u_int number, u_int nargs)
 		name = new_name;
 	} else
 		new_name = NULL;
+
+	/* Correctly decode compat syscalls arguments by stripping the prefix */
+	if (strncmp("freebsd32_", name, strlen("freebsd32_")) == 0) {
+		lookup_name = name + strlen("freebsd32_");
+	} else if (strncmp("freebsd64_", name, strlen("freebsd64_")) == 0) {
+		lookup_name = name + strlen("freebsd64_");
+	} else {
+		lookup_name = name;
+	}
```

Comment:
Maybe we should have libsysdecode able to return the unprefixed version? This is a bit ugly.